packages/autorest.gotest/test/integrationtest/output/signalr/armsignalr/fake_test.go

```diff
@@ -0,0 +1,1718 @@
+		testsuite.Require().True(reflect.DeepEqual(exampleRes, nextResult.OperationList))
+		if err == nil {
+			break
+		}
+	}
+}
+
+func (testsuite *FakeTestSuite) TestSignalR_CheckNameAvailability() {
+	ctx := context.Background()
+	fakeServer := fake.Server{}
+	clientFactory, err := armsignalr.NewClientFactory(testsuite.subscriptionId, testsuite.cred, &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: fake.NewServerTransport(&fakeServer),
+		},
+	})
+	testsuite.Require().NoError(err, "Failed to create client factory")
+	client := clientFactory.NewClient()
+
+	// From example specification/signalr/resource-manager/Microsoft.SignalRService/preview/2021-06-01-preview/examples/SignalR_CheckNameAvailability.json
+	ctx = runtime.WithHTTPHeader(ctx, map[string][]string{
+		"example-id": {"SignalR_CheckNameAvailability"},
+	})
+	var exampleLocation string
+	var exampleParameters armsignalr.NameAvailabilityParameters
+	exampleLocation = "eastus"
+	exampleParameters = armsignalr.NameAvailabilityParameters{
+		Name: to.Ptr("mySignalRService"),
+		Type: to.Ptr("Microsoft.SignalRService/SignalR"),
+	}
+
+	exampleRes := armsignalr.NameAvailability{
+		Message:       to.Ptr("The name is already taken. Please try a different name."),
+		NameAvailable: to.Ptr(false),
+		Reason:        to.Ptr("AlreadyExists"),
+	}
+
+	fakeServer.CheckNameAvailability = func(ctx context.Context, location string, parameters armsignalr.NameAvailabilityParameters, options *armsignalr.ClientCheckNameAvailabilityOptions) (resp azfake.Responder[armsignalr.ClientCheckNameAvailabilityResponse], errResp azfake.ErrorResponder) {
+		testsuite.Require().Equal(exampleLocation, location)
+		testsuite.Require().True(reflect.DeepEqual(exampleParameters, parameters))
+		resp = azfake.Responder[armsignalr.ClientCheckNameAvailabilityResponse]{}
+		resp.SetResponse(http.StatusOK, armsignalr.ClientCheckNameAvailabilityResponse{NameAvailability: exampleRes}, nil)
+		return
+	}
+	res, err := client.CheckNameAvailability(ctx, exampleLocation, exampleParameters, nil)
+	testsuite.Require().NoError(err, "Failed to get result for example specification/signalr/resource-manager/Microsoft.SignalRService/preview/2021-06-01-preview/examples/SignalR_CheckNameAvailability.json")
+	testsuite.Require().True(reflect.DeepEqual(exampleRes, res.NameAvailability))
+}
+
+func (testsuite *FakeTestSuite) TestSignalR_ListBySubscription() {
+	ctx := context.Background()
+	fakeServer := fake.Server{}
+	clientFactory, err := armsignalr.NewClientFactory(testsuite.subscriptionId, testsuite.cred, &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: fake.NewServerTransport(&fakeServer),
+		},
+	})
+	testsuite.Require().NoError(err, "Failed to create client factory")
+	client := clientFactory.NewClient()
```

Comment:
is that possible to init one client from one service only once?

Reply:
same above